src/Adapter/PDF/DeliverySlipPdfGenerator.php

```diff
@@ -58,22 +59,24 @@ public function __construct(TranslatorInterface $translator)
     /**
      * {@inheritdoc}
      */
-    public function generatePDF(array $orderId)
+    public function generatePDF(array $orderIds)
     {
-        if (count($orderId) !== 1) {
-            throw new CoreException(sprintf('"%s" supports generating delivery slip for single order only.', get_class($this)));
+        if (!$orderIds) {
+            throw new MissingDataException('Missing data required to generate PDF');
         }
 
-        $orderId = reset($orderId);
-        $order = new Order((int) $orderId);
+        try {
+            $orders_invoice_collection = new PrestaShopCollection('OrderInvoice');
+            $orders_invoice_collection->where('id_order', 'in', $orderIds);
 
-        if (!Validate::isLoadedObject($order)) {
-            throw new RuntimeException($this->translator->trans('The order cannot be found within your database.', [], 'Admin.Orderscustomers.Notification'));
-        }
+            if (0 === count($orders_invoice_collection)) {
+                throw new RuntimeException($this->translator->trans('The order cannot be found within your database.', [], 'Admin.Orderscustomers.Notification'));
+            }
```

Comment:
Don't understand this point. `OrderIds `can't be empty/null.

Reply:
The PHP array can't be empty indeed. But it doesn't mean all required invoices exist for real in database.
What is supposed to happen if there are less invoices than required orders ? Is it an error case or should we let it go as nominal case ?

Reply:
It's so heavy to check all order on by one (by new Order...). We can compare size of array and throw an error if not exact. But what is the point? To add an information, this PR is inspired by other PDF adapter, example : https://github.com/PrestaShop/PrestaShop/blob/develop/src/Adapter/PDF/CreditSlipPdfGenerator.php. It's not check in CreditSlip.

Reply:
My point here is only about the legal value of delivery slips, and it's only a question of use case to implement.
If it's ok to generate the PDF document for any order whatever its status, then your PR is good enough. Just like the credit slip generator you've linked.
But if all the orders provided are expected to have an invoice available, then the user can tell a few are missing only after going through the entire PDF doc to check manually.

Previous version of the code was straight forward because it could only handle a single order at a time, and threw an exception if the related invoice wasn't found. It would let the user know if called in a simple loop through orders.
But your improvement might silence such behavior if the loop gets replaced with the bulk strategy.

I'm not telling you that's wrong or a mistake, but it might get confusing from a functional use case perspective.
I guess that's the main reason you've been previously asked to tell more about the use case :)

Reply:
We can do a specific use case if `$orderIds == 1`. I thinks it's a better way to do like CreditSlip and the PDF class has a verification on empty data. There is no **single** answer to this problem ;).
But the problem is here, we can't do bulk on this adapter but on others it's possible.

Reply:
There is no single answer to the question indeed. And other adapters don't implement such strict behavior, which is unsafe IMHO.
I suppose the choice should be left to the shop configuration: strict or permissive. There are 3 solutions :
1. Select the behavior with an extra argument in service constructor in order to override the dependency injection in configuration file `generator.yml`
2. Extend this generator into a new class that will implement the strict behavior by default, and select the appropriate implementation in said configuration `generator.yml`
3. Add an extra method `generateStrictPDF` to `PDFGeneratorInterface` to apply the "strict mode", and let the caller pick up the most appriopriate method

Those options have pros and cons, and at the moment such behavior sounds a bit too sophisticated compared to the global design of the product. Perhaps later such "strict mode" will be added to all generators at once, but for now code consistency should prime.
Then please forget about my previous suggestion, and just fix code style in your PR. I will discuss the idea with core team and we'll decide then if strict mode" is required or not :)